tests/components/vegehub/conftest.py

```diff
@@ -0,0 +1,103 @@
+"""Fixtures and test data for VegeHub test methods."""
+
+from collections.abc import Generator
+from dataclasses import dataclass
+from typing import Any
+from unittest.mock import AsyncMock, MagicMock, PropertyMock, patch
+
+import pytest
+from vegehub import VegeHub
+
+from homeassistant.components.vegehub.coordinator import VegeHubCoordinator
+from homeassistant.const import (
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_IP_ADDRESS,
+    CONF_MAC,
+    CONF_WEBHOOK_ID,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, load_fixture
+
+TEST_IP = "192.168.0.100"
+TEST_UNIQUE_ID = "aabbccddeeff"
+TEST_SERVER = "http://example.com"
+TEST_MAC = "A1:B2:C3:D4:E5:F6"
+TEST_SIMPLE_MAC = "A1B2C3D4E5F6"
+TEST_HOSTNAME = "VegeHub"
+TEST_WEBHOOK_ID = "webhook_id"
+HUB_DATA = {
+    "first_boot": False,
+    "page_updated": False,
+    "error_message": 0,
+    "num_channels": 2,
+    "num_actuators": 2,
+    "version": "3.4.5",
+    "agenda": 1,
+    "batt_v": 9.0,
+    "num_vsens": 0,
+    "is_ac": 0,
+    "has_sd": 0,
+    "on_ap": 0,
+}
+
+
+@dataclass
+class VegeHubData:
+    """Define a data class."""
+
+    coordinator: VegeHubCoordinator
+    hub: VegeHub
+
+
+@pytest.fixture
```

Comment:
```suggestion
@pytest.fixture(autouse=True)
```

you probably want that one on autouse, as you'll need it for all tests anyway and then you don't need it in every method signature